synapse/handlers/message.py

```diff
@@ -895,6 +895,21 @@ async def create_and_send_nonmember_event(
 
             spam_check = await self.spam_checker.check_event_for_spam(event)
             if spam_check is not synapse.spam_checker_api.Allow.ALLOW:
+                try:
+                    if isinstance(spam_check, tuple):
+                        [code, dict] = spam_check
+                        raise SynapseError(
+                            403,
+                            "This message had been rejected as probable spam",
+                            code,
+                            dict,
+                        )
+                except ValueError:
+                    logger.warning(
+                        "Spam-check module returned invalid error value. Expecting [code, dict], got %s",
+                        spam_check,
+                    )
+                    spam_check = Codes.FORBIDDEN
```

Comment:
I would prefer something like this:

```
if isinstance(spam_check, tuple):
    try:
        [code, dict] = spam_check
        raise SynapseError(
            403,
            "This message had been rejected as probable spam",
            code,
            dict,
        )
    except ValueError:
        logger.warning(
            "Spam-check module returned invalid error value. Expecting [code, dict], got %s",
            spam_check,
        )
        spam_check = Codes.FORBIDDEN
```

Which doesn't change much functionally but increases the code's readability.

Also I would log it as `error` so that it shows in Sentry etc.